New CMake version

**Describe the bug**
The code does not compile with CMake/3.18.4-GCCcore-10.2.0

**To Reproduce**
Steps to reproduce the behavior:
1. CMake with module load CMake/3.18.4-GCCcore-10.2.0
2. When running make a complaint on:
/software/software/binutils/2.31.1-GCCcore-8.2.0/bin/ld.gold: error: CMakeFiles/NISE.dir/types_MPI.c.o: multiple definition of 'T_NON_TYPE'
/software/software/binutils/2.31.1-GCCcore-8.2.0/bin/ld.gold: CMakeFiles/NISE.dir/NISE.c.o: previous definition here

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/src/types_MPI.h
+++ b/src/types_MPI.h
@@ -14,6 +14,6 @@
 }
 
 typedef CUSTOM_MPI_DATATYPE(60) t_non_datatype;
-const t_non_datatype T_NON_TYPE;
+extern const t_non_datatype T_NON_TYPE;
 
 #endif

